swagger_parser/lib/src/generator/templates/dart_enum_dto_template.dart

```diff
@@ -67,29 +85,71 @@ ${generatedFileComment(
 part '${enumClass.name.toSnake}.mapper.dart';
 
 ${descriptionComment(enumClass.description)}@MappableEnum()
-enum $className {
+enum $className implements Comparable<$className> {
 $values
+$unknownEnumValueStr
+$jsonFieldStr
+$constructorStr
+$toJsonStr
+$toStringStr
+$compareToStr
 }
 ''';
 }
 
-String _constructor(String className) => '\n\n  const $className(this.json);\n';
+String _constructor(String className) =>
+    '\n${indentation(1)} const $className(this.value);';
 
-String _jsonField(UniversalEnumClass enumClass) =>
-    '\n  final ${enumClass.type.toDartType()}? json;';
+String _jsonField(UniversalEnumClass enumClass, bool unknownEnumValue) =>
+    '\n${indentation(1)} final ${enumClass.type.toDartType()}${_hasNullItem(enumClass, unknownEnumValue) ? '?' : ''} value;';
```

Comment:
I don't think it was worth changing the names from json to value

Reply:
It makes it consistent with Dart's own documentation and is representative of what it actually is, since it isn't JSON, it's the value of the enum which can be any supported type.